packages/synapse-react-client/src/components/QueryWrapperPlotNav/RowSetView.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { Suspense, useState } from 'react'
+import { CardContainer } from '../CardContainer/CardContainer'
+import { SynapseTable, SynapseTableConfiguration } from '../SynapseTable'
+import { CardConfiguration } from '../CardContainerLogic'
+import LastUpdatedOn from './LastUpdatedOn'
+import { useDeepCompareMemoize } from 'use-deep-compare-effect'
+import QueryWrapperLoadingScreen from '../QueryWrapper/QueryWrapperLoadingScreen'
+import { SynapseErrorBoundary } from '../error'
+import { useQueryContext } from '../QueryContext'
+import { Box } from '@mui/material'
+import { TablePagination } from '../SynapseTable/TablePagination'
+import { ViewMoreQueryResultsButton } from '../QueryWrapper/ViewMoreQueryResultsButton'
+import { useRowSet } from './UseRowSet'
+
+export type RowSetViewProps = {
+  tableConfiguration?: SynapseTableConfiguration
+  cardConfiguration?: CardConfiguration
+  initialLimit?: number
+  hideDownload?: boolean
+}
+
+/**
+ * Renders a table or cards representing the RowSet. Also displays appropriate pagination controls
+ * depending on the value of QueryContext.isInfinite.
+ * @param props
+ * @constructor
+ */
+export function RowSetView(props: RowSetViewProps) {
+  const { cardConfiguration, hideDownload, initialLimit } = props
+
+  const [initialLimitIsApplied, setInitialLimitIsApplied] = useState(
+    initialLimit != null,
+  )
+
+  const { isInfinite } = useQueryContext()
+  const tableConfiguration = useDeepCompareMemoize(
+    props.tableConfiguration
+      ? {
+          ...props.tableConfiguration,
+          hideDownload,
+        }
+      : undefined,
+  )
+
+  const {
+    rowSet,
+    progressMessage,
+    isLoading,
+    isLoadingNewPage,
+    fetchNextPageOfInfiniteData,
+    hasNextPageOfInfiniteData,
+  } = useRowSet(initialLimitIsApplied ? initialLimit : undefined)
+
+  return (
+    <SynapseErrorBoundary>
+      <div className={`FilterAndView`}>
+        {isLoading && (
+          <QueryWrapperLoadingScreen progressMessage={progressMessage} />
+        )}
+        <Suspense
+          fallback={
+            <QueryWrapperLoadingScreen progressMessage={progressMessage} />
+          }
```

Comment:
Only renders if we have a `rowSet` and one of the children is suspended. I.e. the loader directly above this will never render at the same time as this one

Reply:
Consider adding a comment in code, since this could be confusing to an engineer in the future